doc/extensibility/dataflow_best_practices.md

```diff
@@ -0,0 +1,112 @@
+# DataFlow Best Coding Practices
+These are suggestions to help avoid some common pitfalls when using DataFlow blocks.
+
+## Use Slim DataFlow blocks 
+You should always use the Slim version blocks provided by CPS, unless you need advanced capabilities that are not supported (like processing multiple works at the same time). See [DataFlow Examples](dataflow_example.md) for more information.
+
+## Expose `BroadcastBlock` via API
+You should typically expose `BroadcastBlock` via API instead of TransformBlock, etc
+  - Other blocks will retain data when there is no subscriber
+  - If one subscriber picks a message from a block other than broadcast block, it won't be available to the next
```

Comment:
Is this clearer?

> - A `BroadcastBlock<T>` will forward its messages to _all_ linked target blocks, whereas a `BufferBlock<T>` will only send it to the first target block that accepts it.

Reply:
The limitation is not only about BufferBlock - I think most of the existing blocks have this behavior. 
For example, TransfornBlock is a common block which is a natural choice to process data; if that processed data needs to be exposed it would seem natural to expose the TransformBlock. However that would cause problems so I think it is worth mentioning it.
If you think BufferBlock is also worth calling out, I could include it in that list:
"... via API instead of TransformBlock, BufferBlock, etc"

I like the statement you suggested, I think it is a very good replacement for the 2nd bullet (which I would move to the first place as that is the main reason to use BroadcastBlock).

Reply:
Yes, good point about the other block types. Perhaps this is clearer then:

> - A `BroadcastBlock<T>` will forward its messages to _all_ linked target blocks, whereas most other block types (`BufferBlock<T>`, `TransformBlock<TInput,TOutput>`, etc.) will only send it to the first target block that accepts it.